wgpu-hal/src/gles/device.rs

```diff
@@ -281,9 +281,48 @@ impl super::Device {
                 let glow::ActiveUniform { size, utype, name } =
                     gl.get_active_uniform(program, uniform).unwrap();
 
+                match utype {
+                    glow::INT_SAMPLER_1D
+                    | glow::INT_SAMPLER_1D_ARRAY
+                    | glow::INT_SAMPLER_2D
+                    | glow::INT_SAMPLER_2D_ARRAY
+                    | glow::INT_SAMPLER_2D_MULTISAMPLE
+                    | glow::INT_SAMPLER_2D_MULTISAMPLE_ARRAY
+                    | glow::INT_SAMPLER_2D_RECT
+                    | glow::INT_SAMPLER_3D
+                    | glow::INT_SAMPLER_CUBE
+                    | glow::INT_SAMPLER_CUBE_MAP_ARRAY
+                    | glow::UNSIGNED_INT_SAMPLER_1D
+                    | glow::UNSIGNED_INT_SAMPLER_1D_ARRAY
+                    | glow::UNSIGNED_INT_SAMPLER_2D
+                    | glow::UNSIGNED_INT_SAMPLER_2D_ARRAY
+                    | glow::UNSIGNED_INT_SAMPLER_2D_MULTISAMPLE
+                    | glow::UNSIGNED_INT_SAMPLER_2D_MULTISAMPLE_ARRAY
+                    | glow::UNSIGNED_INT_SAMPLER_2D_RECT
+                    | glow::UNSIGNED_INT_SAMPLER_3D
+                    | glow::UNSIGNED_INT_SAMPLER_CUBE
+                    | glow::UNSIGNED_INT_SAMPLER_CUBE_MAP_ARRAY
+                    | glow::SAMPLER_1D
+                    | glow::SAMPLER_1D_SHADOW
+                    | glow::SAMPLER_1D_ARRAY
+                    | glow::SAMPLER_1D_ARRAY_SHADOW
+                    | glow::SAMPLER_2D
+                    | glow::SAMPLER_2D_SHADOW
+                    | glow::SAMPLER_2D_ARRAY
+                    | glow::SAMPLER_2D_ARRAY_SHADOW
+                    | glow::SAMPLER_2D_MULTISAMPLE
+                    | glow::SAMPLER_2D_MULTISAMPLE_ARRAY
+                    | glow::SAMPLER_2D_RECT
+                    | glow::SAMPLER_2D_RECT_SHADOW
+                    | glow::SAMPLER_3D
+                    | glow::SAMPLER_CUBE
+                    | glow::SAMPLER_CUBE_MAP_ARRAY
+                    | glow::SAMPLER_CUBE_MAP_ARRAY_SHADOW
+                    | glow::SAMPLER_CUBE_SHADOW => continue,
+                    _ => {}
+                }
+
                 if let Some(location) = gl.get_uniform_location(program, &name) {
-                    // Sampler2D won't show up in UniformLocation and the only other uniforms
-                    // should be push constants
```

Comment:
In my modified buffer example, while not a `sampler2d`, a `usampler2d` triggered this code path, that's why there is a match now to guard against sampler types, because they were being included and causing problems later.

Reply:
I wish there was an easier way to detect if this uniform is for a texture. Maybe by name?

Reply:
By the name would be hard since naga names are like _group_0_binding_0, only option would be changing naga's naming scheme